### PR TITLE
Hot fix resource request

### LINKED ISSUE #264: Hot fix Resource Request
chỉnh sửa:
- Thanh tìm kiếm responsive theo laptop và desktop.
- Chỉnh sửa null nếu cv status không có gì.
- Chỉnh sửa UI phù hợp với behavior của người dùng ở resource request khi update các thông số ở resource request cv
- Tối ưu call api

### PR DESCRIPTION
**Pull request by khaletuanncc**
chỉnh sửa:
- Thanh tìm kiếm responsive theo laptop và desktop.
- Chỉnh sửa null nếu cv status không có gì.
- Chỉnh sửa UI phù hợp với behavior của người dùng ở resource request khi update các thông số ở resource request cv
- Tối ưu call api

## Release Notes

- **New Features**
  - Enhanced user interface for CV status and resource request components, including improved layout and responsive design.
  - Added a "No data" message for empty CV status lists.
  - Introduced new styles for better visual hierarchy and responsiveness.
  - Added new properties to manage CV-related data, including a collection for associated CVs.

- **Bug Fixes**
  - Improved error handling and state management in resource request editing.

- **Documentation**
  - Updated CSS and HTML structure for clarity and maintainability.

- **Chores**
  - Optimized data retrieval and control flow in resource request management.

### CODE REVIEW

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.css [hunk 1]
```diff
@@ -153,6 +157,15 @@ table td {
...
 }
 
+.cv-info-width{
+    width: 250px;
+}
+
+.code-style{
+    color: white;
+    background-color: chocolate;
+}
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.css:167]: unnecessary

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 2]
```diff
@@ -134,8 +134,8 @@ export class RequestResourceTabComponent
   public listResourceRequestCV: ResourceRequestCVDto[] = [];
   public isRowExpand: { [key: number]: boolean } = {};
   public expandedRows: number[] = [];
-  public editingRows: { [key: number]: { [key: string]: boolean } } = {};
-  public originalValues: { [index: number]: { [field: string]: any } } = {};
+  public editingRows: { [key: number]: { [key: number]: { [key: string]: boolean } } } = {};
+  public originalValues: { [key: number]: { [index: number]: { [field: string]: any } } } = {};
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts:138]: unnecessary

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 3]
```diff
@@ -271,18 +246,20 @@ export class RequestResourceTabComponent
     this.resourceRequestService.updateKpiPointResourceRequestCV(res).subscribe(
       result => {
         abp.notify.success("Updated KPI Point");
-        this.editingRows[index][field] = false;
+        if (this.editingRows[requestId][index]) {
+          delete this.editingRows[requestId][index][field];
+        }
+        if (this.originalValues[requestId][index]) {
+          this.originalValues[requestId][index][field] = res.kpiPoint;
+        }
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts:258]: ok

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 4]
```diff
@@ -205,64 +205,39 @@ export class RequestResourceTabComponent
...
+        if (this.originalValues[requestId][index]) {
+          this.originalValues[requestId][index][field] = item.cvStatus;
+          const cvStatus = this.cVStatusList.find(res => res.id === item.cvStatusId);
+          const po = this.listRequest.findIndex(res => res.id === requestId);
+          this.listRequest[po].resCV[index][field]  = cvStatus;
         }
       },
       error => {
-        abp.notify.error("Failed to update status");
+        abp.notify.error("Failed to update CV Status");
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts:238]: ok

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 5]
```diff
@@ -291,70 +268,75 @@ export class RequestResourceTabComponent
...
       const po = this.listRequest.findIndex(res => res.id === request.id);
-      this.listRequest[po].resCV[index][field] = this.originalValues[index][field];
-      Object.keys(this.editingRows[index]).forEach(field => {
-        this.editingRows[index][field] = false;
-      });
+      if (this.originalValues[request.id] && this.originalValues[request.id][index]) {
+        this.listRequest[po].resCV[index][field] = this.originalValues[request.id][index][field];
+      }
+      delete this.editingRows[request.id][index][field];
     }
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts]: ok

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 6]
```diff
@@ -205,64 +205,39 @@ export class RequestResourceTabComponent
...
-    }
-    if (!this.originalValues[index][field]) {
-      this.originalValues[index][field] = item[field];
-    }
-    this.editingRows[index][field] = true;
+  edit(index: number, field: string, item: ResourceRequestCVDto, requestId: number): void {
+    this.editingRows = {};
+    this.originalValues = {};
+    this.originalValues[requestId] = { [index]: { [field]: item[field] } };
+    this.editingRows[requestId] = { [index]: { [field]: true } };
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts:212]: It's a behavior that I want

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 7]
```diff
@@ -1253,18 +1238,19 @@ export class RequestResourceTabComponent
...
+  refreshDate(index: number, item: ResourceRequestCVDto, field: string, requestId: number) {
     if(field == 'sendCVDate') {
       item.sendCVDate = null;
-      this.updateSendCVDate(index, item, field);
+      this.updateSendCVDate(index, item, field, requestId);
     }
     else {
       item.interviewDate = null;
-      this.updateInterviewTimeCV(index, item, field);
+      this.updateInterviewTimeCV(index, item, field, requestId);
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts:1252]: just 2 scene

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 8]
```diff
@@ -205,64 +205,39 @@ export class RequestResourceTabComponent
...
-            this.listRequest[index].resCV = rs.result as ResourceRequestCVDto[];
-          }
-        } else {
-          console.error('Unexpected response:', rs);
+        if (this.originalValues[requestId][index]) {
+          this.originalValues[requestId][index][field] = item.cvStatus;
+          const cvStatus = this.cVStatusList.find(res => res.id === item.cvStatusId);
+          const po = this.listRequest.findIndex(res => res.id === requestId);
+          this.listRequest[po].resCV[index][field]  = cvStatus;
         }
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts]: ok

**Review by tiennguyenhuuncc — APPROVED** [review]

**Pull request merged**